src/pool-cl/CLPoolManager.sol

```diff
@@ -177,7 +177,15 @@ contract CLPoolManager is ICLPoolManager, Fees, Extsload {
         vault.accountPoolBalanceDelta(key, delta + feeDelta, msg.sender);
 
         /// @notice Make sure the first event is noted, so that later events from afterHook won't get mixed up with this one
-        emit ModifyLiquidity(id, msg.sender, params.tickLower, params.tickUpper, params.liquidityDelta);
+        emit ModifyLiquidity(
+            id,
+            msg.sender,
+            params.tickLower,
+            params.tickUpper,
+            params.liquidityDelta,
+            feeDelta.amount0(),
```

Comment:
do we need this? if the intend is to know the feeDelta, either

1/ when simulate modifyLiqudiity, it returns feeDelta 

2/ or calculate this value off-chain, via:
a. get the liquidity, feeGrowthInside0LastX128 and feeGrowthInside1LastX128 for this position via `poolManager.getPosition(poolId, owner, tickLower, tickUpper)`

b. get the global fee growth feeGrowthGlobal1X128/feeGrowthGlobal1X128 via `poolmanager.pools(poolId)`

calculate this by doing the math with those value

Reply:
for 1/
yes actually the steps 1&2 in the my notion docs is to simulate modifyLiquidity and get feeDelta results
however, since unlike NonfungiblePositionManager, there are no tokensOwed states for each position in v4-core, we need to somehow record the feesOwed when there are modifyLiquidity(collect) invoked within the epoch

for 2/
that's a more complicated way just as I proposed in my previous version
besides, this method will still require extra event params like feeGrowthInside

also the contract needs to do the computation no matter there is off-chain farming or not, so I don't think emitting the feeDelta results alongside will incur like a unacceptable cost